src/main/java/frc/robot/commands/ramp/ResetRamp.java

```diff
@@ -41,7 +42,10 @@ public void execute() {}
   @Override
   public void end(boolean interrupted) {
     ramp.stopMotor();
-    ramp.resetEncoder();
+    if(ramp.getReversedSwitchState()) {
+      new WaitCommand(Constants.RAMP_WIND_DOWN_TIME);
```

Comment:
This isn't going to do what you want. new WaitCommand() will create a new wait command, yes, but this method does not take any time. It will immediately return and resetEncoder will run.